modules/apps/translation/translation-service/src/main/java/com/liferay/translation/internal/exporter/XLIFF12InfoFormTranslationExporter.java

```diff
@@ -79,33 +84,97 @@ public InputStream exportInfoItemFieldValues(
 
-		Collection<InfoFieldValue<Object>> infoFieldValues =
-			infoItemFieldValues.getInfoFieldValues();
+		Map<String, List<InfoFieldValue<Object>>> infoFieldValuesMap =
+			new LinkedHashMap<>();
+
+		for (InfoFieldValue<Object> infoFieldValue :
+				infoItemFieldValues.getInfoFieldValues()) {
 
-		for (InfoFieldValue<Object> infoFieldValue : infoFieldValues) {
 			InfoField infoField = infoFieldValue.getInfoField();
 
-			if (!_translationInfoFieldChecker.isTranslatable(infoField)) {
-				continue;
+			if (_translationInfoFieldChecker.isTranslatable(infoField)) {
+				List<InfoFieldValue<Object>> infoFieldValuesList =
+					infoFieldValuesMap.computeIfAbsent(
+						infoField.getName(), name -> new ArrayList<>());
+
+				infoFieldValuesList.add(infoFieldValue);
 			}
+		}
+
+		for (Map.Entry<String, List<InfoFieldValue<Object>>> entry :
+				infoFieldValuesMap.entrySet()) {
 
 			Element transUnitElement = bodyElement.addElement("trans-unit");
 
-			transUnitElement.addAttribute("id", infoField.getName());
+			transUnitElement.addAttribute("id", entry.getKey());
 
 			Element sourceElement = transUnitElement.addElement("source");
 
 			sourceElement.addAttribute(
 				"xml:lang", fileElement.attributeValue("source-language"));
+
+			List<InfoFieldValue<Object>> infoFieldValuesList = entry.getValue();
+
+			Stream<InfoFieldValue<Object>> stream =
+				infoFieldValuesList.stream();
+
 			sourceElement.addCDATA(
-				_getStringValue(infoFieldValue.getValue(sourceLocale)));
+				_getStringValue(
+					stream.map(
+						infoFieldValue -> (String)infoFieldValue.getValue(
+							sourceLocale)
+					).collect(
+						Collectors.joining(StringPool.COMMA_AND_SPACE)
+					)));
+
+			if (infoFieldValuesList.size() > 1) {
+				Element segSourceElement = transUnitElement.addElement(
+					"seg-source");
+
+				int mid = 0;
+
+				for (InfoFieldValue<Object> infoFieldValue :
+						infoFieldValuesList) {
+
+					Element mrkElement = segSourceElement.addElement("mrk");
+
+					mrkElement.addAttribute("mid", String.valueOf(mid));
+					mrkElement.addAttribute("mtype", "seg");
+					mrkElement.addCDATA(
+						(String)infoFieldValue.getValue(sourceLocale));
+
+					mid++;
+				}
+			}
 
 			Element targetElement = transUnitElement.addElement("target");
 
 			targetElement.addAttribute(
 				"xml:lang", fileElement.attributeValue("target-language"));
-			targetElement.addCDATA(
-				_getStringValue(infoFieldValue.getValue(targetLocale)));
+
+			if (infoFieldValuesList.size() > 1) {
+				int mid = 0;
+
+				for (InfoFieldValue<Object> infoFieldValue :
+						infoFieldValuesList) {
+
+					Element mrkElement = targetElement.addElement("mrk");
+
+					mrkElement.addAttribute("mid", String.valueOf(mid));
+					mrkElement.addAttribute("mtype", "seg");
+					mrkElement.addCDATA(
+						(String)infoFieldValue.getValue(targetLocale));
+
+					mid++;
+				}
+			}
+			else {
+				InfoFieldValue<Object> infoFieldValue = infoFieldValuesList.get(
```

Comment:
The algorithm for the sourceLocale (line 12-148) and the targetLocale is different for the addCDATA.
It's and intended behavior and why?

Reply:
Yes, it is intended. In XLIFF 1.2, the source tag cannot contain multiple values; so, the source is the concatenation of all present values, and then each individual value is put as a mrk tag inside the source-seg element (XLIFF 1.2 is messy as hell). This doesn't happen with the target, that can either contain the tarteg text as is, or a set of mrk elements. Each mrk element contains only one value.

Reply:
thanks